src/main/java/ch/securify/OutputGenerator.java

```diff
@@ -0,0 +1,142 @@
+        // Loop over contract results
+        for(Map.Entry<String, SolidityResult> contract: allContractResults.entrySet()) {
+            OutputGenerator.handleContract(contract);
+        }
+    }
+
+
+    /**
+     * Handles the output production for a single contract.
+     *
+     * @param a single contract extracted from the contract analysis
+     */
+    private static void handleContract(Map.Entry<String, SolidityResult> contract) {
+        String[] parts = contract.getKey().split(":");
+        String contractPath = parts[0];
+        String contractName = parts[1];
+
+        List<String> contractLines = OutputGenerator.getLines(contractPath);
+
+        for(Map.Entry<String, SmallPatternResult> entry: contract.getValue().results.entrySet()) {
+            OutputGenerator.handlePattern(entry.getKey(), entry.getValue(), contractName, contractPath, contractLines);
+        }
+    }
+
+
+    /**
+     * Handles the output production for a single pattern.
+     *
+     * @param the name of the pattern
+     * @param the actual pattern object containing violations, warnings, conflicts, and safe
+     * @param the name of the contract for which this pattern applies
+     * @param the path of the source file containing the contract for which this pattern applies
+     * @param the content of the source file containing the contract
+     */
+    private static void handlePattern(String patternName, SmallPatternResult pattern, String contractName, String sourceFile, List<String> lines) {
+        for(Integer lineNum: pattern.violations) {
+            String header = OutputGenerator.ANSI_RED + "Violation" + OutputGenerator.ANSI_RESET + " for " + patternName;
+            OutputGenerator.handleLine(contractName, sourceFile, lines, lineNum, header);
+        }
+        for(Integer lineNum: pattern.warnings) {
+            String header = OutputGenerator.ANSI_YELLOW + "Warning" + OutputGenerator.ANSI_RESET + " for " + patternName;
+            OutputGenerator.handleLine(contractName, sourceFile, lines, lineNum, header);
+        }
+        for(Integer lineNum: pattern.conflicts) {
+            String header = OutputGenerator.ANSI_YELLOW + "Warning" + OutputGenerator.ANSI_RESET + " for " + patternName;
+            OutputGenerator.handleLine(contractName, sourceFile, lines, lineNum, header);
+        }
+    }
+
+
+
+    /**
+     * Handles the output production for a single line of a pattern.
+     *
+     * @param the name of the contract to which this line applies
+     * @param the path to the source file containing the contract to which this lines applies
+     * @param the content of the source file containing the contract
+     * @param the line number within the contract to which this lines applies
+     * @param the header to prepend to this line of output
+     */
+    private static void handleLine(String contractName, String sourceFile, List<String> lines, int lineNum, String header) {
+        String output = header;
```

Comment:
See https://stackoverflow.com/questions/5234147/why-stringbuilder-when-there-is-string/5234160#5234160 (not that it really matters here, but just to keep a better code quality).